platforms/software/dependency-management/src/main/java/org/gradle/internal/component/external/model/DefaultModuleComponentArtifactIdentifier.java

```diff
@@ -52,9 +51,7 @@ public String getFileName() {
 
     @Override
     public String getDisplayName() {
-        String name = this.getFileName();
-        String componentIdentifier = this.componentIdentifier.toString();
-        return name + " (" + componentIdentifier + ")";
+        return getFileName() + " (" + getComponentIdentifier().getDisplayName() + ")";
```

Comment:
🤔 So the `getDisplayName()` method doesn't come from the `DisplayName` interface, which provides the `getCapitalizedDisplayName()` method...which is just the display name with a leading capital?  And in many implementations just forwards to `getDisplayName()` without modification.  That's fun.

I don't think `ModuleComponentArtifactIdentifier` needs to implement `DisplayName`, callers of `getCapitalizedDisplayName()` can just call `getDisplayName()` directly.

Reply:
I already went down this path unfortunately. At first glance I agree it does not need to implement `DisplayName`, but the code in `Describables#of` does an `instanceof DisplayName` check. Existing code relies on `ModuleComponentArtifactIdentifier` implementing `DisplayName` even if they are not explicitly calling `getCapitalizedDisplayName`

Reply:
The reason I made `ModuleComponentArtifactIdentifier` implement `DisplayName` was because all the subclasses were as well. This applies to other artifact ID implementations as well. 

I think part of this reason this was done was because we don't really want to capitalize these file names. Even if the user asks for a capitalized name. The capitalized display name is for when you want the display name to start at the beginning of a sentence. However, even if the file is at the beginning of a sentence, we do not want to capitalize the first letter